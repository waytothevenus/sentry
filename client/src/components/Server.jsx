import * as React from "react";

class Server extends React.Component {
	render() {
		return <h1>test</h1>;
	}
}

export default Server;