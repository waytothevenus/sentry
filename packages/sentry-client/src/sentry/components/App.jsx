import * as React from "react";

class App extends React.Component {
	render() {
		return <h1>sentry</h1>;
	}
}

export default connect()(App);