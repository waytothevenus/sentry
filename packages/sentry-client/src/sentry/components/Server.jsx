import * as React from "react";

export class Server extends React.Component {
	render() {
		return <h1>{this.props.hostname}</h1>;
	}
}

export default Server;