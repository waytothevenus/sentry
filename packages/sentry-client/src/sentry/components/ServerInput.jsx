import React from "react";
import { Field, reduxForm } from "redux-form";

export class ServerInput extends React.Component {
	render() {
		const { handleSubmit } = this.props;
		return (
			<form onSubmit={handleSubmit}>
				<Field name="hostname" component="input" type="text" />
				<button type="submit">Submit</button>
			</form>
		);
	}
}

export default reduxForm({ form: "hostnameInput" })(ServerInput);