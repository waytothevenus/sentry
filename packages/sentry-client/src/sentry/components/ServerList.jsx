import * as React from "react";
import { connect } from "react-redux";

export class ServerList extends React.Component {
	render() {
		return <h1>test</h1>;
	}
}



export default connect(ServerList);