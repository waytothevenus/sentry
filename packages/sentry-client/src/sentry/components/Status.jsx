import * as React from "react";

import * as status from "../constants/status";

export class Status extends React.Component {
	render() {
		let className, style;

		switch (this.props.status) {
		case status.STATUS_AVAILABLE:
			className = "fa fa-circle";
			style = { color: "#229926" };
			break;
		case status.STATUS_ISSUE:
			className = "fa fa-exclamation-circle";
			style = { color: "#FEE032" };
			break;
		case status.STATUS_MAINTENANCE:
			className = "fa fa-square";
			style = { color: "#1172C6" };
			break;
		default:
		case status.STATUS_OUTAGE:
			className = "fa fa-exclamation-triangle";
			style = { color: "#951A1D" };
			break;
		}

		return <i className={className} style={style} aria-hidden="true"></i>;
	}
}

export default Status;
