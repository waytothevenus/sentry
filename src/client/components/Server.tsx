import * as React from "react";

class Server extends React.Component<undefined, undefined> {
	render() {
		return <h1>test</h1>;
	}
}

export default Server;