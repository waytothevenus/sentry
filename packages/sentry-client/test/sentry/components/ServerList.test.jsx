import React from "react";
import renderer from "react-test-renderer";
import { ServerList } from "../../../src/sentry/components/ServerList";

test("test ServerList component", () => {
	const component = renderer.create(
		<ServerList/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
