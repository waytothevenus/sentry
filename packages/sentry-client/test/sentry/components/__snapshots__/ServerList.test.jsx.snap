// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ServerList component 1`] = `
<h1>
  test
</h1>
`;
